File: 2025_06/leetcode2025-06-08.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
private:
    vector<int> res;
    int N;

    void dfs(int cur, int is_beg){
        res.push_back(cur);
        for (int i=is_beg; i<=9; i++){
            if (cur*10+i <= N){
                dfs(cur*10+i, is_beg&0);
            }
        }
    }

public:
    vector<int> lexicalOrder(int n) {
        N = n;
        dfs(0, 1);
        res.erase(res.begin());
        return res;
    }
};

int main(){
    Solution s;
    int n = 13;
    vector<int> res = s.lexicalOrder(n);
    for (int i : res){
        cout << i << " ";
    }
    cout << endl;
}

File: 2025_06/leetcode2025-06-08.py
from typing import List

class Solution:
    def lexicalOrder(self, n: int) -> List[int]:
        res = []
        def dfs(cur, is_beg):
            res.append(cur)
            for i in range(is_beg, 10):
                if cur*10+i <= n:
                    dfs(cur*10+i, is_beg&0)
        dfs(0, 1)
        res.pop(0)
        return res

if __name__ == "__main__":
    s = Solution()
    n = 13
    print(s.lexicalOrder(n))
